### DISCUSSION TITLE
Issues with create_model attribute

### DISCUSSION THREAD

**Discussion by ldrlei** [gempy-project/gempy#950] [Q&A] [answered]
Hi everyone, I'll preface my question saying that I am very new with gempy, python and programming in general so bear with me. 
I'm trying to learn GemGIS and Gempy bit by bit, following the basic tutorials (amazing work btw) but I'm stuck on the first model and I'm trying to understand with but to no avail.

I installed GemGIS and Gempy in a new Anaconda environment, as per instruction. I started using Jupyter but changed to Spyder and I followed the steps to create the first model, the planar beds one. 

```python
import pandas as pd
import gempy as gp
import matplotlib.pyplot as plt

interfaces = pd.read_csv('.../gemgis_data-main/data/model1/model1_interfaces.csv', 
                         delimiter = '\t')
interfaces.head()

orientations = pd.read_csv('.../gemgis_data-main/data/model1/model1_orientations.csv', 
                           delimiter='\t')
orientations.head()
```
No problem with importing the modules and with reading the data. 
Then the tutorial says to first create a new empty model with the `create_model` attribute, give it a name and input the data loaded before with the `init_data` attribute. 

```python
geo_model = gp.create_model('model1')
geo_model
```

This is the code I'm using and where I'm stuck, when I run this I'm receiving this error:
`AttributeError: module 'gempy' has no attribute 'create_model'`

I thought maybe I installed something wrong and tried several times with new wnvironments but I always stop here. During my tests I found that the attribute `init_data` gives the same problem. 
Looking around in these threads for a solution I found something like this

```python
geo_model: gp.data.GeoModel = gp.create_geomodel(
    project_name='Model1',
    extent=[0, 150, -10, 10, -100, 0],
    resolution=[100, 2, 100],
    # refinement=6,
    structural_frame=gp.data.StructuralFrame.initialize_default_structure()
)
```

so maybe, I thought, `create_model` and `init_data` were substituted with the `create_geomodel` attribute, because running that does not show me any error. But in that case how can I give the interfaces and orientations data? I cannot pass them as dataframes like with the `init_data` attributes, right? And what is the first part of the code `geo_model: gp.data.GeoModel =` for?

I'm sorry if there are gaps of basic knowledge in my reasoning, I'm trying to learn and solve possible problems along the way but could not find a solution to this one alone. So I'm happy to any correction if I'm referring to things wrongly.

I'm also sorry for the incomplete post, I published the question by mistake while editing and I'm adding info now. I'm seeing already some comments so @javoha let me know if I need to share anything else and @AlexanderJuestel I think that might be the case, I searched around for alternate tutorial on Gempy v3 cause I suspected that could be the cause of my issues but I could not find any...Could I ask where to find them? I surely missed them! 

Thanks a lot for the very quick replies,
Elia

Edit: I think that I found what I was searching for at https://docs.gempy.org/tutorials/ch1_fundamentals/ch1_1_basics.html, I saw the tutorial before but for some reason thought that it was not my case, let me know if I'm wrong. 
Also, I still would like to know the `geo_model: gp.data.GeoModel =` usage if it is possible and could be explained easily. 

Thank you again!

**Comment by javoha** [comment]
Hello @ldrlei,
thanks for your question. Could you please give us a little more information: Which tutorial/model whre you using exactly (link?). In what cell did the error occur and what was the error message?
Cheers,
Jan

**Comment by AlexanderJuestel** [comment]
Hey @ldrlei,

Without knowing your exact issue, I assume that you have installed the latest version of Gempy while the GemGIS tutorials still use the gempy_legacy version. You may want to check out the new Gempy tutorials and adapt the GemGIS tutorials until we have reworked them.

**Comment by javoha** [comment, marked as answer]
Hi @ldrlei ,
thanks for the updated information. So first things first: As @AlexanderJuestel already said - Gempy was updated to a new version a while ago and GemGis has not yet been updated to amtch the new version. The errors you are getting are basically you doing things that do not exist or where renamed in gempy version 3.
What does this mean: Easiest case would be focusing just on gempy right now. You can look at all the tutorials on the gempy page (the one you linked is a nice start) and these should run in your environment. Let me know if you encounter any problems.
For GemGis: It might take a while until it is updated to the new gempy version, so as of now you can not just use the GemGis functionalities. Once you feel a little more familiar with gempy (and coding in general) you might however be able to use some of the functionality or adapt the old GemGis code yourself.
Cheers,
Jan

**Reply by ldrlei** [reply]
Hi @javoha,
thank you very much for your quick response and explanation , that is exactly what I'll do then! 

Cheers, 
Elia

**Comment by DCAwv80** [comment]
I apolgize, I just created an issues about this very subject. I'm having the same issue, but i didn't install GemGIS. I'm assuming then you have to use:
"importer_helper=gp.data.ImporterHelper(
        path_to_orientations=path_to_data+"tutorial_model_orientations.csv",
        path_to_surface_points=path_to_data+"tutorial_model_surface_points.csv"
    )"

My surface points are in a dataframe so I need to convert them to a .csv? we also do not have any orientation points. Is there a way to tell it there are no orientation points? from the errors I get it looks like you have to have both for this to work.

Finaly, it would be nice to see the .csv files mentioned because there is no link to download these at the bottom ot the tutorial. 

Any guidance is much appreciated!

Thank you for your time!
Dan Amrine

**Reply by javoha** [reply]
See #988
